platform/core.network/src/org/netbeans/core/network/proxy/pac/impl/NbPacScriptEvaluator.java

```diff
@@ -196,6 +200,8 @@ public class NbPacScriptEvaluator implements PacScriptEvaluator {
     private static final String PAC_SOCKS5_FFEXT = "SOCKS5"; // Mozilla Firefox extension. Not part of original Netscape spec.
     private static final String PAC_HTTP_FFEXT = "HTTP"; // Mozilla Firefox extension. Not part of original Netscape spec.
     private static final String PAC_HTTPS_FFEXT = "HTTPS"; // Mozilla Firefox extension. Not part of original Netscape spec.
+    private static class RPSingleton { private static final RequestProcessor instance = new RequestProcessor(NbPacScriptEvaluator.class.getName(), Runtime.getRuntime().availableProcessors(), true, false); }
+    private static RequestProcessor getRequestProcessor() { return RPSingleton.instance; }
```

Comment:
Why this complex construct? Why not just

```
private static final RequestProcessor RP = new RequestProcessor(NbPacScriptEvaluator.class.getName(), Runtime.getRuntime().availableProcessors(), true, false);
```

This is done all over the code base. The proposed style looks overly complex without clear benefits.

Why the limit to number available processors?  For a real PAC script I would expect it to be throttled by network IO (DNS look ups for example). If high CPU usage is the problem (and thus an attacker is expected), I would go with a throughput of 1.

Reply:
Lazy initialisation based on the value of ProxySettings.pacScriptTimeout

Reply:
I too am not convinced that this would gain anything. Since RP itself is lazy. All you safe is the RP object and few internal fields of it.

btw the least-bad version of this idiom is JDK 17+.

change to `javac.release=17` in project properties
```java
    private static RequestProcessor getRequestProcessor() {
        class Lazy {
            static final RequestProcessor RP = new RequestProcessor(NbPacScriptEvaluator.class.getName(), Runtime.getRuntime().availableProcessors(), true, false);
        }
        return Lazy.RP;
    }
```
keeps the hack local. (also please don't write everything in one line)

Reply:
Made the changes and squashed the commits. @mbien Please approve the PR

Reply:
@subhash-arabhi thank you for the update. However you never answered the question: "Why the limit to number available processors?". Why does the intended throughput scale with the number or processors? From my POV that makes no sense to me.